sda-commons-server-prometheus/src/test/java/org/sdase/commons/server/prometheus/PrometheusBundleTest.java

```diff
@@ -199,6 +202,8 @@ void micrometerMetricsAvailableInPrometheus() {
     assertThat(sampleList.get(0).labelValues).contains("testTagValue");
 
     assertThat(sampleList.get(0).value).isEqualTo(2);
+
+    Metrics.globalRegistry.close();
```

Comment:
Would it also be possible to close the registry in the Bundle? Or is [`removeRegistry`](https://github.com/SDA-SE/sda-dropwizard-commons/pull/2484/files#diff-c7d888d64ab52c9b42637100cad05ce6be7c9d9b6d9e81811ad2317c9e8e5f23R94) doing exactly that?
Would it make sense (and is it possible) to also close (or reset) the `CollectorRegistry` on shutdown?

Reply:
No closing, clearing and removing have different effects:

- Metrics.globalRegistry.clear();
   Clears all meters (e.g. Counter) but keeps them. registries unaffected

- Metrics.globalRegistry.close();
   Keeps registries but does not accept new meters and does not publish new values

- Metrics.removeRegistry(meterRegistry);
   actually removes the registry

Currently doing some tests to see the effect of doing both removingRegistry and closing

Reply:
Although it should be possible to use `CollectorRegistry.defaultRegistry.clear();` the side effects are hard to predict. We are using CollectorRegistry.defaultRegistry.unregister(collector) in multiple places. If we clear the registry first and then unregister there will be a null pointer exceptions.

Given the impact of this issue i suggest we analyse the clearing of the prometheus registry separately.